pipeline/src/pipeline/do_inference.jl

```diff
@@ -17,3 +17,24 @@ function do_inference(truthdata, pipeline::AbstractEpiAwarePipeline)
         truthdata, inference_configs, pipeline; tspan, inference_method)
     return inference_results
 end
+
+"""
+Generate inference results using the specified truth data and pipeline. This is
+example mode.
+
+# Arguments
+- `truthdata`: The truth data used for generating inference results.
+- `pipeline::EpiAwareExamplePipeline`: Sets the pipeline behavior.
+
+# Returns
+An array of inference results.
+
+"""
+function do_inference(truthdata, pipeline::EpiAwareExamplePipeline)
```

Comment:
Am I missing something? Why can't this rely on the abstract/default method?

Reply:
The thing this method does differently is pick an inference config randomly out of the list generated by `make_inference_configs`. An alternate approach would be to do the dispatch at that level though (connected to #289 )

Reply:
This seems like it would reduce copy and paste across methods?